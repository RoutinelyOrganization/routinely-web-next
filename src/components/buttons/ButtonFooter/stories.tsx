import ButtonFooter from '../ButtonFooter';

export default {
  title: 'buttons/ButtonFooter',
  component: ButtonFooter,
};

export const Template = () => (
  <div>
    <ButtonFooter />
  </div>
);

Template.parameters = {
  backgrounds: { default: 'white' },
};
