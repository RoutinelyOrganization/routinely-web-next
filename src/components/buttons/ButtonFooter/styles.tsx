'use client';

import media from '@/styles/mediaQueries';
import { colors } from '@/styles/variables';
import styled from 'styled-components';

export const ButtonFooter = styled.button`
  position: fixed;
  bottom: 0;
  width: 100%;
  background-color: ${colors.primary};
  border: none;
  padding: 8px;
  display: none;
  cursor: pointer;
  z-index: 20;

  > img {
    width: 59px;
    height: 38px;
    display: block;
    margin: auto;
  }

  ${media.mobile} {
    display: block;
  }
`;
