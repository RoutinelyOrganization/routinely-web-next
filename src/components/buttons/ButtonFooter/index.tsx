import * as S from './styles';

export default function ButtonFooter({
  children,
  ...props
}: React.ButtonHTMLAttributes<HTMLButtonElement>) {
  return (
    <S.ButtonFooter {...props} role="button" data-testid="button-footer">
      {children}
    </S.ButtonFooter>
  );
}
