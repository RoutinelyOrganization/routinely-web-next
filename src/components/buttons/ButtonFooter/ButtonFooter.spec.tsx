import { render } from '@testing-library/react';
import ButtonFooter from '.';

describe('ButtonFooter Component', () => {
  it('should render according to the snapshot', () => {
    const { container } = render(<ButtonFooter />);
    expect(container).toMatchSnapshot();
  });
});
