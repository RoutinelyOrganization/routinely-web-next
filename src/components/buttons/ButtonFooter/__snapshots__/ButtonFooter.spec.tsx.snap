// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonFooter Component should render according to the snapshot 1`] = `
.c0 {
  position: fixed;
  bottom: 0;
  width: 100%;
  background-color: #5C59BB;
  border: none;
  padding: 8px;
  display: none;
  cursor: pointer;
  z-index: 20;
}

.c0 >img {
  width: 59px;
  height: 38px;
  display: block;
  margin: auto;
}

@media (max-width: 500px) {
  .c0 {
    display: block;
  }
}

<div>
  <button
    class="c0"
    data-testid="button-footer"
    role="button"
  >
    <img
      alt="BotÃ£o de inicio"
      src="[object Object]"
    />
  </button>
</div>
`;
